src/drug_discovery/structures/protein.py

```diff
@@ -785,6 +785,94 @@ def to_pdb(self, file_path: str):
                 f"Failed to write structure to file {file_path}: {str(e)}"
             ) from e
 
+    @beartype
+    def to_base64(self) -> str:
+        """Convert the protein to base64 encoded PDB format.
+
+        Returns:
+            str: Base64 encoded string of the PDB file content
+        """
+        import base64
+        import tempfile
+
+        # Create a temporary PDB file
+        with tempfile.NamedTemporaryFile(
+            mode="w", suffix=".pdb", delete=False
+        ) as temp_file:
+            temp_file_path = temp_file.name
+
+        try:
+            # Write the protein to the temporary file
+            self.to_pdb(temp_file_path)
+
+            # Read the file and encode to base64
+            with open(temp_file_path, "rb") as f:
+                pdb_content = f.read()
+                base64_encoded = base64.b64encode(pdb_content).decode("utf-8")
+
+            return base64_encoded
+        finally:
+            # Clean up the temporary file
+            import os
+
+            if os.path.exists(temp_file_path):
+                os.remove(temp_file_path)
+
+    @classmethod
+    @beartype
+    def from_base64(
+        cls,
+        base64_string: str,
+        name: str = "",
+        **kwargs: Any,
+    ) -> "Protein":
+        """
+        Create a Protein instance from a base64 encoded PDB string.
+
+        Args:
+            base64_string (str): Base64 encoded PDB content
+            name (str, optional): Name of the protein. Defaults to "".
+            **kwargs: Additional arguments to pass to the constructor
+
+        Returns:
+            Protein: A new Protein instance
+
+        Raises:
+            DeepOriginException: If the base64 string cannot be decoded or parsed
+        """
+        import base64
+        import tempfile
+
+        try:
+            # Decode the base64 string
+            pdb_content = base64.b64decode(base64_string)
+
+            # Create a temporary file with the decoded content
+            with tempfile.NamedTemporaryFile(
+                mode="wb", suffix=".pdb", delete=False
+            ) as temp_file:
+                temp_file.write(pdb_content)
+                temp_file_path = temp_file.name
+
+            # Create the protein from the temporary PDB file
+            protein = cls.from_file(temp_file_path, **kwargs)
```

Comment:
If from_file raises, the temp file leaks. Use a finally with contextlib.suppress(FileNotFoundError) for deletion in both functions.